build_tools/cmake/iree_macros.cmake

```diff
@@ -252,3 +252,26 @@ function(iree_add_executable_dependencies EXECUTABLE DEPENDENCY)
     add_dependencies(${EXECUTABLE} ${DEPENDENCY})
   endif()
 endfunction()
+
+#-------------------------------------------------------------------------------
+# Tests
+#-------------------------------------------------------------------------------
+
+# iree_add_test_environment_properties
+#
+# Adds test environment variable properties based on the current build options.
+#
+# Parameters:
+# TEST_NAME: the test name, e.g. iree/base:ref_ptr_test
+# TARGET_NAME: the target name, e.g. iree_base_ref_ptr_test
+function(iree_add_test_environment_properties TEST_NAME TARGET_NAME)
+  # Can technically build just a target backend and run tests which don't
+  # require a corresponding HAL driver, but that's an edge case - just disable
+  # if either isn't being built.
```

Comment:
I don't think it really is an edge case. That's like, the whole compiler.

Reply:
For binaries yes, for tests that use this awkward filtering I'm less convinced.

There aren't many / any tests in the runtime that use this filtering. The HAL drivers are light enough to build if you want to run associated compiler tests.

There are compiler tests which just use the compiler (`iree-opt`) and tests which use the compiler and runtime (`iree-run-mlir`), and both use this filtering mechanism without differentiating or qualifying their usage. Until that changes, I'd be okay with this setup.

Reply:
The iree-run-mlir tests that use this use it because they don't support the driver. It was introduced to allow build bot coverage where we don't have vulkan support. There are no iree-opt tests that use this:

https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/google/iree%24+IREE_VULKAN_DISABLE+iree-opt&patternType=regexp

Reply:
My concern is not so much that turning off a driver turns of the compiler tests for that driver (doesn't seem quite right, but meh), but rather that doing it this way means making more tests get excluded by the same mechanism that excludes based on driver presence, which means we lose CI coverage and even when we get the CI coverage (working on that) lose the ability to turn off tests for drivers. The drivers are quite fiddly, the backends mostly not.

Reply:
Fixed one Vulkan test that was newly excluded. Do you still have concerns about this?

Reply:
I think this can now be updated to only look at the driver variables.

Reply:
Why? If you enable a driver but not the compiler target, the tests will fail.

Reply:
Ah right.... I think this comment isn't quite right though. We're only using this for iree-run-mlir tests that require both. I'd prefer to limit the spread of this and avoid people using it to disable other tests.

Reply:
Elaborated on the comment.